### PR TITLE
nixos/sudo: fix `security.sudo.package`

### PR DESCRIPTION
**Pull request by nicoonoclaste**
## Description of changes

Fix buggy handling of `security.sudo.package`, where the specified package would be used for
the suid wrappers (`sudo` and `sudoedit`) but not for `systemPackages`, causing pkgs' `sudo`
to be used instead for many executables (incl. `visudo`), the manpages, etc.

This bug has been present since the introduction of that option in d2dc0ae2033d1655a2011fc932bca5cd9c7bfe5b (Oct 2020)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
- [x] Tested, as applicable:
  - NixOS test `sudo`
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Shawn8901** [comment]
Possibly we can also remove line 9 inherit of the sudo package. I did just a short scroll over the file, but it looked quite unused now (which is good 😊).

**Review by bjornfor — APPROVED** [review]

**Review by the-furry-hubofeverything — APPROVED** [review]
briefly skimming it, LGTM

File: nixos/modules/security/sudo.nix [hunk 1]
```diff
@@ -247,7 +245,7 @@ in
       };
     };
 
-    environment.systemPackages = [ sudo ];
+    environment.systemPackages = [ cfg.package ];
```
Comment (pbsds) [inline nixos/modules/security/sudo.nix:248]: Considering we have the SUID wrappers, is sudo in `systemPackages` needed at all?

Comment (Ma27) [reply nixos/modules/security/sudo.nix:248]: At the very least for man-pages & localized messages, perhaps for sudoers.so et al., but that's enough reason already to not discuss it in this bugfix PR. If you'd like to investigate further, feel free :)

**Review by Ma27 — APPROVED** [review]
Good catch, thanks!

**Pull request merged**